src/ArduinoIoTCloud.cpp

```diff
@@ -182,14 +200,18 @@ void ArduinoIoTCloudClass::poll()
   update();
 }
 
-void ArduinoIoTCloudClass::update()
+void ArduinoIoTCloudClass::update(CallbackFunc onSyncCompleteCallback)
 {
   // If user call update() without parameters use the default ones
-  update(MAX_RETRIES, RECONNECTION_TIMEOUT);
+  update(MAX_RETRIES, RECONNECTION_TIMEOUT, onSyncCompleteCallback);
 }
 
-void ArduinoIoTCloudClass::update(int const reconnectionMaxRetries, int const reconnectionTimeoutMs)
+void ArduinoIoTCloudClass::update(int const reconnectionMaxRetries, int const reconnectionTimeoutMs, CallbackFunc onSyncCompleteCallback)
 {
+  unsigned long const timestamp = getTimestamp(); 
+  //check if a property is changed 
+  if(timestamp) Thing.updateTimestampOnChangedProperties(timestamp);
```

Comment:
This `if` statement works because any integer value != 0 is interpreted as true in C/C++- however, it's not very expressive. I'd suggest the following change:
```suggestion
  if(timestamp != 0) Thing.updateTimestampOnChangedProperties(timestamp);
```
Or a even more expressive version could look like that:
```C++
bool const is_valid_timestamp = timestamp != 0;
if(is_valid_timestamp) ...
```
Now you give the reader of the code an idea what the `if` condition is actually checking for.